controllers/workspace/provision/deployment.go

```diff
@@ -140,6 +140,31 @@ func DeleteWorkspaceDeployment(ctx context.Context, workspace *devworkspace.DevW
 	return true, nil
 }
 
+// ScaleDeploymentToZero scales the cluster deployment to zero
+func ScaleDeploymentToZero(ctx context.Context, workspace *devworkspace.DevWorkspace, client runtimeClient.Client) error {
+	defaultRetry := time.Millisecond * 100
+	timeout := time.Second * 60
+	err := wait.Poll(defaultRetry, timeout, func() (done bool, err error) {
+		clusterDeployment, err := getClusterDeployment(common.DeploymentName(workspace.Status.WorkspaceId), workspace.Namespace, client)
+		if err != nil {
+			return false, err
+		}
+		if clusterDeployment == nil {
+			return false, nil
```

Comment:
Shouldn't this be `true, nil`? `getClusterDeployment` will only return `nil, nil` if it gets an `isNotFound` error from the cluster, i.e. if the deployment doesn't exist. It looks like this will spin for 60 seconds if a deployment does not exist.

Reply:
Yeah that makes sense, i've removed the polling and so this should be fixed!